adapters/adair-uae.js

```diff
@@ -0,0 +1,309 @@
+/**
+ * This code is responsible for implementing all methods related to fetching
+ * and returning data from Abu Dhabi in the United Arab Emirates
+ *
+ * This is a two-stage adapter requiring loading multiple urls before parsing
+ * data.
+ */
+'use strict';
+
+import { REQUEST_TIMEOUT } from '../lib/constants';
+import { default as baseRequest } from 'request';
+import _ from 'lodash';
+import { default as moment } from 'moment-timezone';
+import async from 'async';
+
+const request = baseRequest.defaults({timeout: REQUEST_TIMEOUT});
+
+exports.name = 'adair-uae';
+
+/**
+ * Fetches the data for a given source and returns an appropriate object
+ * @param {object} source A valid source object
+ * @param {function} cb A callback of the form cb(err, data)
+ */
+exports.fetchData = function (source, cb) {
+  // Fetch both the measurements and meta-data about the locations
+  var tasks = [];
+  _.forEach(stations, function (s) {
+    const sourceName = source.url + s.urlName;
+    var task = function (cb) {
+      request({
+        'rejectUnauthorized': false,
```

Comment:
As Joe mentioned, I think it's a good idea to remove `rejectUnauthorized: false` and include the certs just to be safe. I just did it for EEA (in PR #729) and it's pretty straightforward. 1) Get the certs you need (you can do this using [open_ssl](https://www.npmjs.com/package/ssl-root-cas) or even your browser, clicking on the lock next to the url in Firefox will give you the whole certificate chain) 2) Include it in the `/certs` folder 3) Include a line injecting the certificate, like [this](https://github.com/openaq/openaq-fetch/blob/79f6ca16fbbe9c0a22b1c13d49a21d53245c7068/adapters/eea-direct.js#L12)

Reply:
Okay I added what I assume is the right certificate for the fetching sites, and I believe I did it the same way @sruti, explained, but it still does not seem to work, could I get help?

Reply:
Looking into this now @magsyg!